extensions/aws/s3/S3WrapperBase.cpp

```diff
@@ -144,7 +144,7 @@ bool S3WrapperBase::deleteObject(const std::string& bucket, const std::string& o
 int64_t S3WrapperBase::writeFetchedBody(Aws::IOStream& source, const int64_t data_size, const std::shared_ptr<io::BaseStream>& output) {
   static const uint64_t BUFFER_SIZE = 4096;
   std::vector<uint8_t> buffer;
-  buffer.reserve(BUFFER_SIZE);
+  buffer.resize(BUFFER_SIZE);
```

Comment:
Both the static bufffer size and the preallocation adds performance overhead.
```c++
std::vector<unsigned char> buffer(4096);
```

Reply:
Updated in [48ec7a](https://github.com/apache/nifi-minifi-cpp/pull/979/commits/48ec7a2ebb6d109c147f2a3b83aa2ae8996acfa2)